src/main/java/com/netchill/services/moviepreview/MoviePreviewService.java

```diff
@@ -0,0 +1,85 @@
+package com.netchill.services.moviepreview;
+
+import com.netchill.api.moviedb.models.Preview;
+import com.netchill.api.moviedb.models.MovieDbPaginatedResponse;
+import com.netchill.api.moviedb.MovieDbApiClient;
+import com.netchill.db.dao.movie.MovieDao;
+import com.netchill.db.generated.Genre;
+import com.netchill.services.configuration.ConfigurationService;
+import com.querydsl.core.Tuple;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.text.DecimalFormat;
+import java.util.List;
+
+@Singleton
+public class MoviePreviewService {
+    private final MovieDbApiClient movieDbApiClient;
+    private final MovieDao movieDao;
+    private final ConfigurationService configurationService;
+
+    @Inject
+    private MoviePreviewService(MovieDbApiClient movieDbApiClient, MovieDao movieDao, ConfigurationService configurationService) {
+        this.movieDbApiClient = movieDbApiClient;
+        this.movieDao = movieDao;
+        this.configurationService = configurationService;
+    }
+
+    private void updateImageUrl(List<Preview> previews){
+        String imgKey;
+        for(int i = 0; i < previews.size(); i++){
+            imgKey = previews.get(i).getBackdrop_path();
```

Comment:
tu peux faire un foreach :

```suggestion
        for(Preview preview:  previews.size()){
            imgKey = preview.getBackdrop_path();
```